app/src/main/java/org/mozilla/tv/firefox/MainActivity.kt

```diff
@@ -179,6 +180,26 @@ class MainActivity : LocaleAwareAppCompatActivity(), OnUrlEnteredListener, Media
         }
 
         if (fragmentManager.backStackEntryCount > 0) {
+            // TODO: remove this when FocusRepo is in place #1395
+            val maybePocketFragment = (supportFragmentManager.findFragmentByTag(PocketVideoFragment.FRAGMENT_TAG)
+                    as PocketVideoFragment?)?.let {
+                if (it.isVisible) it else null
+            }
+
+            val maybeSettingsFragment = (supportFragmentManager.findFragmentByTag(SettingsFragment.FRAGMENT_TAG)
+                    as SettingsFragment?)?.let {
+                if (it.isVisible) it else null
+            }
+
+            val overlayFragment = (supportFragmentManager.findFragmentByTag(NavigationOverlayFragment.FRAGMENT_TAG)
+                    as NavigationOverlayFragment)
+
+            if (maybePocketFragment != null) {
+                overlayFragment.defaultFocusTag = PocketVideoFragment.FRAGMENT_TAG
+            } else if (maybeSettingsFragment != null) {
+                overlayFragment.defaultFocusTag = SettingsFragment.FRAGMENT_TAG
+            }
```

Comment:
I think it would be a readability improvement (and also easier to refactor out later) if we moved extracted this focus logic to a separate method.